app/public/src/game.js

```diff
@@ -0,0 +1,87 @@
+(function(){'use strict';}());
+
+function Game(){
+  var frames, gameOver, totalScore, allowedNoOfFrames;
+  this.frames = [];
+  this.gameOver = false;
+  this.totalScore = 0;
+  this.allowedNoOfFrames = 10;
+}
+
+Game.prototype.newGame = function(){
+  this.frames = [];
+};
+
+Game.prototype.addFrame = function(frameScore){
+  if (this.frames.length === this.allowedNoOfFrames){
+    throw new Error("Game over");
+  }
+    this.frames.push(frameScore);
+    if (this.frames.length === this.allowedNoOfFrames){
+      this.gameOver = true;
+    }
+};
+
+Game.prototype.showFrame = function(frameNo){
+  return this.frames[frameNo-1];
+};
+
+Game.prototype.currentRollNumber = function(){
+ return parseInt(this.frames.length);
+}
+
+Game.prototype.showAllFrames = function(){
+  return this.frames;
+};
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXX Roll XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+function Roll(){
+  var roll, rollComplete, score, strikeScore, spareScore, subScore;
+  this.roll = [];
+  this.rollComplete = false;
+  this.score = 0;
```

Comment:
a frame only has only one score and it's derived from some number of bowls; it's not clear why you would need a separate property for score, let alone four.